app/src/main/java/org/zalando/nakadi/filters/LoggingFilter.java

```diff
@@ -179,13 +203,157 @@ private void logToAccessLog(final RequestLogInfo requestLogInfo, final int statu
                 timeSpentMs,
                 requestLogInfo.contentEncoding,
                 requestLogInfo.acceptEncoding,
-                requestLogInfo.contentLength,
-                responseLength);
+                requestLogInfo.getRequestLength(),
+                requestLogInfo.getResponseLength());
     }
 
     private boolean isPublishingRequest(final RequestLogInfo requestLogInfo) {
         return requestLogInfo.path != null && "POST".equals(requestLogInfo.method) &&
                 requestLogInfo.path.startsWith("/event-types/") &&
                 (requestLogInfo.path.endsWith("/events") || requestLogInfo.path.endsWith("/events/"));
     }
+
+    // ====================================================================================================
+    private static class RequestWrapper extends HttpServletRequestWrapper {
+
+        private CountingInputStreamWrapper inputStreamWrapper;
+
+        RequestWrapper(final HttpServletRequest request) {
+            super(request);
+        }
+
+        long getInputStreamBytesCount() {
+            return inputStreamWrapper != null ? inputStreamWrapper.getCount() : 0;
+        }
+
+        @Override
+        public ServletInputStream getInputStream() throws IOException {
+            if (inputStreamWrapper == null) {
+                inputStreamWrapper = new CountingInputStreamWrapper(super.getInputStream());
+            }
+            return inputStreamWrapper;
+        }
+    }
+
+    private static class CountingInputStreamWrapper extends ServletInputStream {
+
+        private final ServletInputStream wrappedInputStream;
+        private final CountingInputStream countingInputStream;
+
+        CountingInputStreamWrapper(final ServletInputStream wrappedInputStream) {
+            this.wrappedInputStream = wrappedInputStream;
```

Comment:
do we need because there is not implementation of some of the methods in CountingInputStream ?

Reply:
Yes, this allows us to delegate calls to `isReady()` and `set*Listener` to the wrapped `ServletInputStream`.
I've copied the code from here initially: https://github.com/zalando/nakadi/blob/master/core-common/src/main/java/org/zalando/nakadi/util/CompressionBodyRequestFilter.java#L119

There we are using a narrower `InputStream` as the parameter type, which doesn't let us use the delegation for these methods.